packages/core/src/node-license/getPromoCode.ts

```diff
@@ -23,6 +23,43 @@ export interface PromoCode {
 export async function getPromoCode(promoCode: string): Promise<PromoCode> {
     const provider = getProvider();
     const nodeLicenseContract = new ethers.Contract(config.nodeLicenseAddress, NodeLicenseAbi, provider);
+
+    
     const promoCodeDetails = await nodeLicenseContract.getPromoCode(promoCode);
-    return promoCodeDetails;
+    const promoCodeIsActive = promoCodeDetails[1];
+
+    // If the promo code is active, return the details
+    if(promoCodeIsActive){
+        return promoCodeDetails;
+    }
+    
+    // Otherwise Check if the promo code is an address that owns a node license
+    // If this is the case, it means the promo code has not been used yet
+    // This will return the zero address if the promo code is not an address
+    const promoCodeAsAddress = await nodeLicenseContract.validateAndConvertAddress(promoCode);
+
+    // If the promo code is an address
+    if(promoCodeAsAddress!== ethers.ZeroAddress){ 
```

Comment:
Is this necessary ? We can locally check if its [a valid address](https://docs.ethers.org/v5/api/utils/address/#utils-isAddress) and just check the balance right ?

Reply:
This has been changed!